### PR TITLE
[Website] update check-links

### PR DESCRIPTION
**Pull request by bullet03**
- update check-links.sh script
- add func to find absolute links, leading to prod and staging
- add manually verified list with links that should be ignored
- sort invalid links by error code status
------------------------
before:
before_ch
after:
- checking absolute production links
after_ch_1
after_ch_3
- checking staging links
after_ch_2
after_ch_4
- invalid links sorted by error status code
after_ch_5
------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by bullet03** [comment]
R: @aromanenko-dev, could you kindly help with review?

**Review by aromanenko-dev — CHANGES_REQUESTED** [review]
Please, check the naming of methods and variable names.

Do we run this script automatically? What is an example of he output?

File: website/www/check-links.sh [hunk 1]
```diff
@@ -67,25 +99,54 @@ mapfile -t external_links < <(printf '%s\n' "${links[@]}" | grep "^https\?://" |
...
-done
-# Clear line - hide progress bar
-echo -n -e "\033[2K"
+function handleUrls {
+  i=1
+  for external_link in "${external_links[@]}"
+  do
+      redraw_progress_bar 50 1 $i ${#external_links[@]}
+
+      curlResult=$(curl -sSfL --max-filesize 1000000 --max-time 10 --connect-timeout 10 --retry 2 -4 "$external_link" 2>&1 > /dev/null) && status=$? || status=$?
```
[the remark this answers is not in the document]
Comment (bullet03) [reply website/www/check-links.sh]: changed

File: website/www/check-links.sh [hunk 2]
```diff
@@ -67,25 +99,54 @@ mapfile -t external_links < <(printf '%s\n' "${links[@]}" | grep "^https\?://" |
...
 
-    if ! curl -sSfL --max-filesize 1000000 --max-time 10 --connect-timeout 10 --retry 2 -4 "${external_link}" > /dev/null ; then
-        invalid_links+=("${external_link}")
-        echo "${external_link}"
-    fi
-    i=$((i+1))
-done
-# Clear line - hide progress bar
-echo -n -e "\033[2K"
+function handleUrls {
```
[the remark this answers is not in the document]
Comment (bullet03) [reply website/www/check-links.sh]: > nit: use the same naming style format, rename to `handle_urls`

changed

**Comment by bullet03** [comment]
> Please, check the naming of methods and variable names.
> 
> Do we run this script automatically? What is an example of he output?

- Yes, script is run automatically
- Example of the output:




Example of the output is also provided in the description of the PR.

**Review by aromanenko-dev — APPROVED** [review]
LGTM

**Pull request merged**